### DISCUSSION TITLE
Support bulb Heiman 7w rgb led

### DISCUSSION THREAD

**Discussion by d3m10n** [Koenkk/zigbee2mqtt#15812] [Devices]
I have a unknown bulb „Ampoule LED Zigbee Ampoule Couleur E27 Ampoule Connectée 7W Ampoule Intelligente Zigbee“ from [Amazon](https://www.amazon.fr/Connectée-Intelligente-Changement-Dimmable-Décoration/dp/B096M5X2CY/ref=sr_1_1?__mk_fr_FR=ÅMÅŽÕÑ&crid=3HPYVFS3HLR28&keywords=Heiman+7w+rgb&qid=1672354428&sprefix=heiman+7w+rgb%2Caps%2C63&sr=8-1). A set Switch Heiman HS2WDSR-E + Bulb

After i added this to a extern  converter, the bulb changed to „supported“ and all works fine:

`const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['ColorLight'],
    ZigbeeManufacturer: ['HEIMAN'],
    model: 'ColorLight',
    vendor: 'Heiman',
    description: 'Heimanlamp',    
    extend: extend.light_onoff_brightness_colortemp_color(),
};

module.exports = definition; `

The bulb:

``availability:
  - topic: zigbee2mqtt/bridge/state
brightness: true
brightness_scale: 254
color_mode: true
command_topic: zigbee2mqtt/Lampe1 /set
device:
  identifiers:
    - zigbee2mqtt_0x00158d000
  manufacturer: Heiman
  model: Heimanlamp (ColorLight)
  name: 'Lampe1 '
  sw_version: 2000-1002
effect: true
effect_list:
  - blink
  - breathe
  - okay
  - channel_change
  - finish_effect
  - stop_effect
json_attributes_topic: 'zigbee2mqtt/Lampe1 '
max_mireds: 500
min_mireds: 150
name: 'Lampe Table manger '
schema: json
state_topic: 'zigbee2mqtt/Lampe1 '
supported_color_modes:
  - xy
  - color_temp
unique_id: 0x00158d000_light_zigbee2mqtt
platform: mqtt ``

**Comment by Krull56** [comment]
Thanks ! I have the same model and it works now ;)

**Reply by d3m10n** [reply]
For changing Colors with the 5button Switch nodered:
`[{"id":"0871acfb35494f51","type":"tab","label":"Flow 2","disabled":false,"info":"","env":[]},{"id":"459d6658bfa638ff","type":"function","z":"0871acfb35494f51","name":"function 2","func":"// function for color choose (right/left button) \n// heiman switch HS2WDSR\n\n// read xy_color \nvar action_color_x = msg.payload.action_color.x;\nvar action_color_y = msg.payload.action_color.y;\nvar action_transition_time = msg.payload.action_transition_time;\n\n\n// Set message\nmsg.payload = [\n    {\n        \"data\":\n        {\n            \"xy_color\": [action_color_x, action_color_y],\n            \"transition\": action_transition_time\n        }\n    }\n];\n\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":180,"y":200,"wires":[["4bdbfc0c1417f37d","e744d2a9e2aca1b3"]]},{"id":"40c4ef276753fc94","type":"mqtt in","z":"0871acfb35494f51","name":"","topic":"zigbee2mqtt/Dimmer Lampe manger","qos":"2","datatype":"json","broker":"7303fce2ae3dcee7","nl":false,"rap":true,"rh":0,"inputs":0,"x":160,"y":80,"wires":[["0fbd41e36637dd24","ffe84a6a323922c5"]]},{"id":"0fbd41e36637dd24","type":"debug","z":"0871acfb35494f51","name":"debug 1","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":500,"y":20,"wires":[]},{"id":"e091b15c964ed8bb","type":"debug","z":"0871acfb35494f51","name":"debug 2","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":520,"y":340,"wires":[]},{"id":"4bdbfc0c1417f37d","type":"debug","z":"0871acfb35494f51","name":"debug 3","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":540,"y":160,"wires":[]},{"id":"e744d2a9e2aca1b3","type":"api-call-service","z":"0871acfb35494f51","name":"Light table manger couleur","server":"%%ID%%","version":5,"debugenabled":false,"domain":"light","service":"turn_on","areaId":[],"deviceId":[],"entityId":["light.lampe_table_manger"],"data":"payload[0].data","dataType":"jsonata","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":460,"y":240,"wires":[["e091b15c964ed8bb"]]},{"id":"ffe84a6a323922c5","type":"switch","z":"0871acfb35494f51","name":"","property":"payload.action","propertyType":"msg","rules":[{"t":"eq","v":"color_move","vt":"str"}],"checkall":"true","repair":false,"outputs":1,"x":390,"y":80,"wires":[["459d6658bfa638ff"]]},{"id":"7303fce2ae3dcee7","type":"mqtt-broker","name":"Local mqtt","broker":"mqtt://core-mosquitto:1883","port":"1883","clientid":"","autoConnect":true,"usetls":false,"protocolVersion":"4","keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthRetain":"false","birthPayload":"","birthMsg":{},"closeTopic":"","closeQos":"0","closeRetain":"false","closePayload":"","closeMsg":{},"willTopic":"","willQos":"0","willRetain":"false","willPayload":"","willMsg":{},"userProps":"","sessionExpiry":""},{"id":"%%ID%%","type":"server","name":"Home Assistant","addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"","connectionDelay":false,"cacheJson":false,"heartbeat":false,"heartbeatInterval":"","areaSelector":"id","deviceSelector":"id","entitySelector":"id","statusSeparator":"","statusYear":"hidden","statusMonth":"hidden","statusDay":"hidden","statusHourCycle":"default","statusTimeFormat":"h:m","enableGlobalContextStore":false}] `

**Comment by mtgpcs** [comment]
Where do I add the

"The bulb:" section to? I got the bulb working in Z2M but not sure where that section goes.
